ipywidgets/widgets/widget_dom_events.py

```diff
@@ -0,0 +1,77 @@
+from .widget_core import CoreWidget
+from .domwidget import DOMWidget
+from .trait_types import InstanceDict
+from .widget import register, widget_serialization, CallbackDispatcher
+from traitlets import Unicode, List, Bool, validate
+
+
+@register
+class DOMListener(CoreWidget):
+    _model_name = Unicode('DOMListenerModel').tag(sync=True)
+    source = InstanceDict(DOMWidget).tag(sync=True, **widget_serialization)
+    watched_events = List().tag(sync=True)
+    ignore_modifier_key_events = Bool(False).tag(sync=True)
+    prevent_default_action = Bool(False).tag(sync=True)
+    _supported_mouse_events = List([
+        'click',
+        'auxclick',
+        'dblclick',
+        'mouseenter',
+        'mouseleave',
+        'mousedown',
+        'mouseup',
+        'mousemove',
+        'wheel',
+        'contextmenu',
+        'dragstart',
+        'drag',
+        'dragend',
+        'dragenter',
+        'dragover',
+        'dragleave',
+        'drop'
+    ]).tag(sync=True)
+    _supported_key_events = List([
+        'keydown',
+        'keyup'
+    ]).tag(sync=True)
+
+    def __init__(self, **kwargs):
+        super(DOMListener, self).__init__(**kwargs)
+        self._dom_handlers = CallbackDispatcher()
+        self.on_msg(self._handle_mouse_msg)
+
+    @property
+    def supported_key_events(self):
+        return self._supported_key_events
+
+    @property
+    def supported_mouse_events(self):
+        return self._supported_mouse_events
+
+    @validate('watched_events')
+    def _validate_watched_events(self, proposal):
+        value = proposal['value']
+        bad_events = [v for v in value if v not in
+                      self._supported_mouse_events
+                      + self._supported_key_events]
+        if bad_events:
+            raise ValueError('The event(s) {} are not '
+                             'supported.'.format(bad_events))
```

Comment:
I always like it when error msg'es say what the option are, for instance, in matplotlib if I don't know the marker type, i just type nonsense and i'll pick something from the error msg, beats reading documentation :)